### PR TITLE
feature/WDP190703-11

### PR DESCRIPTION
**Pull request by danielmark0116**
Style the main-menu for RWD compatibility.

Doing so, I set breakpoints variables (the same values as in the bootstrap docs) and added mixins for phone view, tablet view and desktops view

I also cherry picked the style.scss, variables.scss and  mixins folder from previous task

### CODE REVIEW

**Comment by danielmark0116** [comment]
But the desktop mixin is probably useless as Im thinking now :)

File: src/index.html [hunk 1]
```diff
@@ -21,5 +21,6 @@
 
     <script src="./scripts/bootstrap.min.js"></script>
     <script src="./scripts/app.js"></script>
+    <script src="./scripts/main-menu.js"></script>
```
Comment (matwjablonski) [inline src/index.html]: why you don't import this js into app js?

File: src/sass/components/_header.scss [hunk 2]
```diff
@@ -221,5 +221,107 @@ header {
...
+        &.show {
+          height: auto;
+        }
+      }
+
+      ul {
+        flex-direction: column;
+        width: 100%;
+
+        li {
```
Comment (matwjablonski) [inline src/sass/components/_header.scss]: don't use tags for styling

File: src/scripts/main-menu.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// FUNCTION TRIGGERING MAIN MENU on phone screens
+
+function openMenu () {
+  document.querySelector('#main-menu').classList.toggle('show');
+  document.querySelector('#main-menu-btn').classList.toggle('active');
+}
+
+// CLICK LISTENER
+document.querySelector('#main-menu-btn').addEventListener('click', function () {
```
Comment (matwjablonski) [inline src/scripts/main-menu.js]: `querySelector` method is much slower than `getElementById`

File: src/scripts/main-menu.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+// FUNCTION TRIGGERING MAIN MENU on phone screens
+
+function openMenu () {
+  document.querySelector('#main-menu').classList.toggle('show');
+  document.querySelector('#main-menu-btn').classList.toggle('active');
+}
+
+// CLICK LISTENER
```
Comment (matwjablonski) [inline src/scripts/main-menu.js]: this comment is saying nothing - remove it

File: src/scripts/main-menu.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+document.querySelector('#main-menu-btn').addEventListener('click', function () {
+  openMenu();
```
Comment (matwjablonski) [inline src/scripts/main-menu.js]: you can write it simpler:

```js
const menuBtn = document.getElementById('main-menu-btn');

if (menuBtn) {
  menuBtn.addEventListener('click', openMenu);
}
```

File: src/index.html (hunk 1 above)
Comment (danielmark0116) [reply src/index.html]: Fixed :)

File: src/sass/components/_header.scss (hunk 2 above)
Comment (danielmark0116) [reply src/sass/components/_header.scss]: Fixed :)

File: src/scripts/main-menu.js (hunk 3 above)
Comment (danielmark0116) [reply src/scripts/main-menu.js]: Fixed :)

File: src/scripts/main-menu.js (hunk 4 above)
Comment (danielmark0116) [reply src/scripts/main-menu.js]: Fixed :)

File: src/scripts/main-menu.js (hunk 5 above)
Comment (danielmark0116) [reply src/scripts/main-menu.js]: Fixed :)

File: src/scripts/app.js [hunk 6]
```diff
@@ -1 +1,2 @@
-console.log('hello');
+// IMPORT -> Main Menu Btn trigger on phone screens
+require('./main-menu');
```
Comment (matwjablonski) [inline src/scripts/app.js]: import / not require

File: src/scripts/main-menu.js [hunk 7]
```diff
@@ -1,11 +1,9 @@
-// FUNCTION TRIGGERING MAIN MENU on phone screens
+const menuBtn = document.getElementById('main-menu-btn');
 
+// FUNCTION TRIGGERING MAIN MENU on phone screens
```
Comment (matwjablonski) [inline src/scripts/main-menu.js]: comment to remove

File: src/scripts/main-menu.js [hunk 8]
```diff
@@ -1,11 +1,9 @@
...
 function openMenu () {
   document.querySelector('#main-menu').classList.toggle('show');
   document.querySelector('#main-menu-btn').classList.toggle('active');
 }
 
-// CLICK LISTENER
-document.querySelector('#main-menu-btn').addEventListener('click', function () {
-  openMenu();
-});
+module.exports = menuBtn && menuBtn.addEventListener('click', openMenu);
```
Comment (matwjablonski) [inline src/scripts/main-menu.js]: without `module.exports`

File: src/scripts/app.js (hunk 6 above)
Comment (danielmark0116) [reply src/scripts/app.js]: Fixed :)

File: src/scripts/main-menu.js (hunk 7 above)
Comment (danielmark0116) [reply src/scripts/main-menu.js]: Fixed :)

File: src/scripts/main-menu.js (hunk 8 above)
Comment (danielmark0116) [reply src/scripts/main-menu.js]: Fixed :)

Comment (danielmark0116) [reply src/scripts/main-menu.js]: Fixed :)

**Comment by matwjablonski** [comment]
resolve conflicts

**Comment by danielmark0116** [comment]
Conflicts resolved

File: src/scripts/main-menu.js [hunk 9]
```diff
@@ -0,0 +1,8 @@
+const menuBtn = document.getElementById('main-menu-btn');
+
+function openMenu () {
+  document.querySelector('#main-menu').classList.toggle('show');
```
Comment (matwjablonski) [inline src/scripts/main-menu.js]: change querySelecter to getElementById ;)

Comment (danielmark0116) [reply src/scripts/main-menu.js]: Fixed :)

File: src/sass/components/_header.scss [hunk 10]
```diff
@@ -221,5 +221,107 @@ header {
...
+      .container > .row {
+        height: auto;
+      }
+
+      .menu-list-container {
+        width: 100%;
+        height: 1px;
+        overflow-y: hidden;
+        padding-top: 1px;
+        &.show {
```
Comment (thisisui) [inline src/sass/components/_header.scss]: to be consistent put enter before this line

Comment (danielmark0116) [reply src/sass/components/_header.scss]: Fixed :)

**Comment by danielmark0116** [comment]
Conflicts resolved, had to merge two app.js imports

**Pull request merged**